src/reportSources/legacyFflogs/eventAdapter/sortStatus.ts

```diff
@@ -0,0 +1,70 @@
+import {getDataArrayBy} from 'data'
+import {getActions} from 'data/ACTIONS'
+import {StatusKey, getStatuses} from 'data/STATUSES'
+import {Event} from 'event'
+import _ from 'lodash'
+import {AdapterStep} from './base'
+
+interface EventNode {
+	event: Event
+	isRoot: boolean
+	after: Event[]
+}
+
+export class SortStatusAdapterStep extends AdapterStep {
+	override postprocess(adaptedEvents: Event[]): Event[] {
+		const sortedEvents: Event[] = []
+		let currentEvents: Event[] = []
+
+		for (const event of adaptedEvents) {
+			if (currentEvents.length === 0 || event.timestamp === currentEvents[0].timestamp) {
+				currentEvents.push(event)
+			} else {
+				sortedEvents.push(...this.sortTimestampBucket(currentEvents))
+				currentEvents = [event]
+			}
+		}
+
+		sortedEvents.push(...this.sortTimestampBucket(currentEvents))
+
+		return sortedEvents
+	}
+
+	private canActionApplyStatus = _.memoize((action: number, status: number) => {
+		const statusKey = _.findKey(getStatuses(this.report), other => status === other.id) as StatusKey | undefined
+
+		// This shouldn't be possible, but bail if we don't find a matching status
+		if (statusKey == null) { return false }
+
+		const applyingActions = getDataArrayBy(getActions(this.report), 'statusesApplied', statusKey)
+
+		return applyingActions.some(other => action === other.id)
+	})
+
+	private sortTimestampBucket(events: Event[]): Event[] {
+		const nodes: EventNode[] = events.map(event => ({
+			event: event,
+			isRoot: true,
+			after: [],
+		}))
+
+		for (const [index, node] of nodes.entries()) {
+			const event = node.event
+
+			if (event.type === 'statusApply') {
+				// Sorts the event after its corresponding action event
+				for (const other of nodes.slice(index + 1)) {
+					if (other.event.type === 'action' && this.canActionApplyStatus(other.event.action, event.status)) {
```

Comment:
while absolutely _miniscule_ in likelyhood of ever happening, i _think_ this has a minor logic error;

if two actors use the same action at precisely the same time, then both `statusApply` events will be pushed to both `action`s

to be double sure, you'll probably want to
- check that the action and status events come from the same source
- `break` after pushing it to make sure wild edge cases won't cause double pushes of the same status apply event

Reply:
👍 good catch, I believe these edge cases are handled now.